### ISSUE TITLE
Privacy violation: auto-accepting subscription requests

### ISSUE THREAD

**Issue by ge0rg** [KaidanIM/Kaidan#220]
Kaidan is silently and automatically accepting subscription requests. This is a violation of the user's privacy, because it leaks the user's availability status (allowing to monitor their behavior) and exposing full JIDs, allowing to target individual clients.

As today most spam sending bots first send a subscription request and then a message, and many spam filters block messages from strangers, you are also exposing users to spam.

Server-side logs:

```
Mar 23 03:00:30 s2sin56532061a490       debug   Received[s2sin]: <presence type='subscribe' to='georg@yax.im' from='redirect59926@jabber.o
zerki.net'>                              
Mar 23 03:00:30 s2sout56530f8819b0      debug   sending: <presence type='unavailable' to='redirect59926@jabber.ozerki.net' from='georg@yax
.im'>                                    
Mar 23 03:00:30 c2s5653115ea470 debug   Received[c2s]: <presence type='subscribed' to='redirect59926@jabber.ozerki.net' from='georg@yax.im/e57cc71c-e400-4a4d-a60b-513ec9d4a880'>  
```

Final result:
![image](https://user-images.githubusercontent.com/165635/37817729-18826b96-2e78-11e8-83ba-8854247bea3e.png)

If you want to improve the UX without exposing the user, I recommend the following:
* [XEP-0379: Pre-Authenticated Roster Subscription](https://xmpp.org/extensions/xep-0379.html)
* [XEP-0401: Easy User Onboarding](https://xmpp.org/extensions/xep-0401.html)

**Comment by LNJ2** [comment]
I agree with you, this should definitely changed. The auto subscription thing still comes from very early versions of kaidan. I already had this on my personal list, but I wasn't aware of that spam bots use it, so it's even more important.

I also wanted to split contacts and chat lists, so you can chat with others who are not in your contacts at all.

Also btw, the icons are missing due to a kirigami bug. It was already fixed but the new version hasn't been added to debian yet. To fix it you could compile kaidan using qmake (which will then use kirigami from a submodule).

**Comment by LNJ2** [comment]
I'm not sure if that's possible at all with the normal gloox API. I can either return true or false; there's no 'don't handle'. But this won't be for a long time, I'll add the GUI soon™.

**Comment by mwild1** [comment]
So what will you do in the UI? Show a modal dialog that blocks the whole client?

**Comment by LNJ2** [comment]
For now, yes. Later I might change this, but it will probably require me to adapt the library.

**Comment by ge0rg** [comment]
Just a small remark that I can't do further testing until this issue is properly sorted out (I'm using the account productively, and I've got a bunch of other clients perfectly capable of roster management, and I can't have Kaidan interfere with them).

**Comment by mwild1** [comment]
@LNJ2 so this would cause the connection to timeout if the user did not accept/decline fast enough, because the gloox event loop would be blocked and the client will not be able to respond to connection events (e.g. pings from the server, or any incoming messages).

I think you need give this some thought, because it's not a trivial matter, and is fundamental to how you design the client's internal architecture.

**Comment by LNJ2** [comment]
Yeah, you're right. I think gloox is more a xmpp bot  library in this point. I'll think about it.

**Comment by LNJ2** [comment]
@mwild1 I just looked into the documentation and saw that it's not correct what you've said, even if I first also thought so. The thing is that you can just enable asynchronous subscription handling, which will eliminate these problems.
